### DISCUSSION TITLE
Add PostValidateBlueprint as ContentController action

### DISCUSSION THREAD

**Discussion by mcl-sz** [umbraco/Umbraco-CMS#15560] [Features and ideas]
Since the security update of Umbraco 8.18.12 it is no longer possible to access the PostSaveBlueprint without the user having access to the "Settings" section.

Although this is understandable, it has a negative effect on the commonly used DocTypeGrid editor. This uses PostSaveBlueprint to validate content.

It would be nice if an extra endpoint (for example PostValidateBlueprint) was added to the ContentController so that a blueprint is validated, possibly with validation whether the user has access to either the content section or the settings section.

See also #https://github.com/umbraco/Umbraco-CMS/issues/15460#issuecomment-1884869057

**Comment by bergmania** [comment]
We don't plan to add new features for v8. 
I guess the main challenge of making it yourselves in the DTGE package is the fact that the ContentSaveValidationAttribute is internal right?

I think the best you can do for now is to make your own validation attribute, that potentially uses reflection to execute the logic from Umbraco.

**Comment by Emetico** [comment]
> I guess the main challenge of making it yourselves in the DTGE package is the fact that the ContentSaveValidationAttribute is internal right?

It's not only the ContentSaveValidationAttribute that is internal. Its also ContentSaveModelValidator, ContentModelValidator, BlueprintItemBinder, ContentItemBinder and ContentModelBinderHelper. Then you would also need some of the logic from the ContentController methods like SetupBlueprint, EnsureUniqueName, PostSaveBlueprint, PostSaveInternal, SaveAndNotify, AddVariantValidationError, GetVariantName, HandleInvalidModelState, MapValuesForPersistence, MapToDisplay and ofcause some of the logic from the abstract ContentControllerBase that the ContentController inherit from.

**Comment by nikolajlauridsen** [comment]
This has been addressed by ##15638 so I'll go ahead and close this discussion
